### PR TITLE
Improvements to BitSet performance and benchmarks

### PR DESCRIPTION
**Pull request by johnynek**
This is a WIP. A core change is using an `Array[Int]` vs `Array[Long]` internally. This helped a bit on jvm and we presume will be a big win on scala-js.

### CODE REVIEW

**Comment by johnynek** [comment]
cc @non

**Comment by larsrh** [comment]
This looks like it has more commits than it should have. Did you forget to rebase to master?

File: bench/src/main/scala/BitSetBenchmarks.scala [hunk 1]
```diff
@@ -1,131 +1,426 @@
...
-
 import scala.collection.{immutable, mutable}
 import scala.math.pow
 import scala.util.Random
 
 @State(Scope.Benchmark)
 @BenchmarkMode(Array(Mode.AverageTime))
 @OutputTimeUnit(TimeUnit.MICROSECONDS)
-class BitSetBenchmarks {
+abstract class DoSetup {
```
[the remark this answers is not in the document]
Comment (anicolaspp) [reply bench/src/main/scala/BitSetBenchmarks.scala]: 👍

**Review by anicolaspp — DISMISSED** [review]

File: bench/src/main/scala/BitSetBenchmarks.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (anicolaspp) [reply bench/src/main/scala/BitSetBenchmarks.scala]: @johnynek what happened with the build on this PR?

**Comment by anicolaspp** [comment]
Let's make sure this PR builds as it should

**Review by broughammer — DISMISSED** [review]

**Comment by osleonard** [comment]
@johnynek is there still a plan for this PR ? Or shall we close it ?

**Comment by johnynek** [comment]
looks like scalastyle foiled the design of the benchmark code previously:
```
/home/travis/build/typelevel/cats-collections/bench/src/main/scala/BitSetBenchmarks.scala: Number of types declared in the file exceeds 30
```

I'll merge with master and make sure that issue is fixed, and I'll share the benchmarks run against head and with this PR so we can see if the wins are worth it.

**Comment by johnynek** [comment]
I picked this back up and merged and ran the benchmarks:
```
Array[Int]

[info] Benchmark                         (range)  (size)  Mode  Cnt   Score   Error  Units
[info] CcbsBenchmarks.buildCcbs           262144    1024  avgt    5   8.691 ± 0.048  us/op
[info] CcbsBenchmarks.bulkAddCcbs         262144    1024  avgt    5   8.002 ± 0.021  us/op
[info] CcbsBenchmarks.bulkAddCcbs2        262144    1024  avgt    5  11.597 ± 0.060  us/op
[info] CcbsBenchmarks.bulkRemoveCcbs2     262144    1024  avgt    5  12.447 ± 0.018  us/op
[info] CcbsBenchmarks.diffCcbs            262144    1024  avgt    5   3.776 ± 0.109  us/op
[info] CcbsBenchmarks.foldIntoCcbs        262144    1024  avgt    5  52.329 ± 0.553  us/op
[info] CcbsBenchmarks.foldOutOfCcbs       262144    1024  avgt    5  10.642 ± 0.918  us/op
[info] CcbsBenchmarks.intersectCcbs       262144    1024  avgt    5   3.610 ± 0.018  us/op
[info] CcbsBenchmarks.intersectTestCcbs   262144    1024  avgt    5   0.493 ± 0.004  us/op
[info] CcbsBenchmarks.lookupCcbs          262144    1024  avgt    5   0.840 ± 0.001  us/op
[info] CcbsBenchmarks.mergeCcbs           262144    1024  avgt    5   3.780 ± 0.214  us/op
[info] CcbsBenchmarks.xorCcbs             262144    1024  avgt    5   3.325 ± 0.034  us/op

master:
[info] Benchmark                         (range)  (size)  Mode  Cnt   Score   Error  Units
[info] CcbsBenchmarks.buildCcbs           262144    1024  avgt    5  34.102 ± 0.139  us/op
[info] CcbsBenchmarks.bulkAddCcbs2        262144    1024  avgt    5  10.614 ± 0.032  us/op
[info] CcbsBenchmarks.bulkRemoveCcbs2     262144    1024  avgt    5  10.635 ± 0.033  us/op
[info] CcbsBenchmarks.diffCcbs            262144    1024  avgt    5   2.268 ± 0.017  us/op
[info] CcbsBenchmarks.foldIntoCcbs        262144    1024  avgt    5  55.141 ± 0.309  us/op
[info] CcbsBenchmarks.foldOutOfCcbs       262144    1024  avgt    5   4.382 ± 0.018  us/op
[info] CcbsBenchmarks.intersectCcbs       262144    1024  avgt    5   2.036 ± 0.006  us/op
[info] CcbsBenchmarks.intersectTestCcbs   262144    1024  avgt    5   0.552 ± 0.001  us/op
[info] CcbsBenchmarks.lookupCcbs          262144    1024  avgt    5   0.942 ± 0.001  us/op
[info] CcbsBenchmarks.mergeCcbs           262144    1024  avgt    5   2.090 ± 0.103  us/op
[info] CcbsBenchmarks.xorCcbs             262144    1024  avgt    5   2.110 ± 0.275  us/op
```
on my M1, generally the performance is actually slightly worse, so I think the motivation of the PR is gone. I've pushed the latest branch in case someone ever wanted to pick it back up.

**Pull request closed without merging**